### ISSUE TITLE
Compiler API `transpileModule` does not emit diagnostics ever - empty array

### ISSUE THREAD

**Issue by alexweej** [microsoft/TypeScript#51962]
# Bug Report

### 🔎 Search Terms

diagnostics empty 

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because I was in a rush and assume this is just a user error

### ⏯ Playground Link

Not usable

### 💻 Code

```
alex@weejmbp ts-test % cat test.ts                    
import * as ts from "typescript";

// function test(a: string, b: number) {
//   return b.startsWith(a);
// }

const result = ts.transpileModule(
  `
function test(a: string, b: number) {
    return b.startsWith(a);
}
`,
  {
    reportDiagnostics: true,
    compilerOptions: {
      strict: true,
      target: ts.ScriptTarget.ES2020,
      noImplicitAny: true,
      noEmitOnError: true,
    },
  }
);

console.log("Result:", result);

alex@weejmbp ts-test % npx tsc test.ts && node test.js
Result: {
  outputText: '"use strict";\nfunction test(a, b) {\n    return b.startsWith(a);\n}\n',
  diagnostics: [],
  sourceMapText: undefined
}
```

### 🙁 Actual behavior

The diagnostics is always an empty array. It doesn't matter how broken the source is, whether it's Type level errors or severe syntactic malformedness.

### 🙂 Expected behavior

Diagnostics provided so I can report the code issues.

----

Thanks for your help!

**Comment by andrewbranch** [comment]
Type errors are not reported in `transpileModule`; that’s the whole point of the API. I can’t reproduce a lack of syntax error reporting though. When I run your test program and modify the input by removing the `c` from `function` to be `funtion test(`..., this is the result:

```
❯ node dist/a.js
Result: {
  outputText: '"use strict";\n' +
    'funtion;\n' +
    'test(a, string, b, number);\n' +
    '{\n' +
    '    return b.startsWith(a);\n' +
    '}\n',
  diagnostics: [
    {
      file: [SourceFileObject],
      start: 1,
      length: 7,
      messageText: "Unknown keyword or identifier. Did you mean 'function'?",
      category: 1,
      code: 1435,
      reportsUnnecessary: undefined
    },
    {
      file: [SourceFileObject],
      start: 15,
      length: 1,
      messageText: "',' expected.",
      category: 1,
      code: 1005,
      reportsUnnecessary: undefined
    },
    {
      file: [SourceFileObject],
      start: 26,
      length: 1,
      messageText: "',' expected.",
      category: 1,
      code: 1005,
      reportsUnnecessary: undefined
    },
    {
      file: [SourceFileObject],
      start: 36,
      length: 1,
      messageText: "';' expected.",
      category: 1,
      code: 1005,
      reportsUnnecessary: undefined
    }
  ],
  sourceMapText: undefined
}
```

**Comment by alexweej** [comment]
@andrewbranch, thanks for replying so quickly! I must have done something wrong when testing broken source code - I was laughing at how remarkable the effort was that TypeScript had gone to to turn utter nonsense into valid JS...!

Which API is recommended for compiling and emitting diagnostics (is this the wrong term for type errors?) in one go? Similar to how a human running 'tsc' will get compiled results and type errors.

Thanks

**Comment by andrewbranch** [comment]
There’s an (old, but mostly alright) example here: https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API#a-minimal-compiler

**Comment by alexweej** [comment]
Thank you, Andrew. I'll give this a go. Looks like this probably is just a PEBKAC issue!

**Comment by boltex** [comment]
@alexweej Long shot here: I'm writing a web application that tries to compile typescript into javascript and I 've got the same problem how to have transpileModule to output diagnostics ? It seems to always give out valid code and no errors no matter horrible the code I put in...? Did the minimal compiler example contain anything specific to be done in a browser ?
